bgpd/bgp_route.c

```diff
@@ -2487,6 +2487,7 @@ bool subgroup_announce_check(struct bgp_dest *dest, struct bgp_path_info *pi,
 		     && IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local))
 		    || (!reflect && !transparent
 			&& IN6_IS_ADDR_LINKLOCAL(&peer->nexthop.v6_local)
+			&& IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local)
```

Comment:
Are you saying that the next-hop length should be set to 32 bytes only if MP_REACH_NLRI (`mp_nexthop_local`) exists? Otherwise, send only a global address?

What if the interface does not have a global address at all? In such a case global address can be just `::` or link-local also. Valid pairs: `::(LL)`, `LL(LL)`, `GUA`, `GUA(LL)`.

Reply:
I'm going to close this PR as bad attempt to fix.  I am not sure how to articulate the fix along side all other valid cases :-(